sdk/eventhub/azure-eventhubs/azure/eventhub/eventprocessor/event_processor.py

```diff
@@ -71,148 +72,154 @@ def __init__(self, eventhub_client: EventHubClient, consumer_group_name: str,
+
         self._consumer_group_name = consumer_group_name
         self._eventhub_client = eventhub_client
         self._eventhub_name = eventhub_client.eh_name
         self._partition_processor_factory = partition_processor_factory
         self._partition_manager = partition_manager
         self._initial_event_position = kwargs.get("initial_event_position", "-1")
+        # TODO: initial position provider will be a callable
+        #  so users can create initial event position for every partition
         self._max_batch_size = eventhub_client.config.max_batch_size
         self._receive_timeout = eventhub_client.config.receive_timeout
-        self._tasks = []  # type: List[asyncio.Task]
+        self._polling_interval = kwargs.get("polling_interval", 10)
+        self._ownership_timeout = self._polling_interval * 2  # TODO: Team haven't decided if this is a separate argument
+        self._tasks = {}  # type: Dict[str, asyncio.Task]
         self._id = str(uuid.uuid4())
+        self._running = False
 
     def __repr__(self):
         return 'EventProcessor: id {}'.format(self._id)
 
     async def start(self):
         """Start the EventProcessor.
 
-        1. retrieve the partition ids from eventhubs.
-        2. claim partition ownership of these partitions.
-        3. repeatedly call EvenHubConsumer.receive() to retrieve events and call user defined PartitionProcessor.process_events().
+        1. Calls the OwnershipManager to keep claiming and balancing ownership of partitions in an
+        infinitely loop until self.stop() is called.
+        2. Cancels tasks for partitions that are no longer owned by this EventProcessor
+        3. Creates tasks for partitions that are newly claimed by this EventProcessor
+        4. Keeps tasks running for partitions that haven't changed ownership
+        5. Each task repeatedly calls EvenHubConsumer.receive() to retrieve events and call user defined partition processor
 
         :return: None
 
         """
         log.info("EventProcessor %r is being started", self._id)
-        partition_ids = await self._eventhub_client.get_partition_ids()
-        claimed_list = await self._claim_partitions(partition_ids)
-        await self._start_claimed_partitions(claimed_list)
+        ownership_manager = OwnershipManager(self, self._eventhub_client, self._ownership_timeout)
+        if not self._running:
+            self._running = True
+            while self._running:
+                claimed_ownership_list = await ownership_manager.claim_ownership()
+                if claimed_ownership_list:
+                    claimed_partition_ids = [x["partition_id"] for x in claimed_ownership_list]
+                    to_cancel_list = self._tasks.keys() - claimed_partition_ids
+                    self._create_tasks_for_claimed_ownership(claimed_ownership_list)
+                else:
+                    log.warning("EventProcessor %r hasn't claimed an ownership. It keeps claiming.", self._id)
```

Comment:
Should the log level be as high as warning?

Reply:
changed to info level, which makes more sense.